src/components/FormField/SelectContext.tsx

```diff
@@ -0,0 +1,101 @@
+import {
+  createContext,
+  HTMLAttributes,
+  ReactElement,
+  ReactNode,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+
+const SelectValue = (props: HTMLAttributes<HTMLDivElement>) => <div {...props} />;
+
+type ContextProps = {
+  selectedValue?: string | null;
+  onSelect: (value: string) => void;
+  popperOpen: boolean;
+  valueNode: ReactNode | null;
+  onOpenChange: (value: boolean) => void;
+};
+
+const SelectContext = createContext<ContextProps>({
+  selectedValue: undefined,
+  onSelect: () => null,
+  popperOpen: false,
+  valueNode: null,
+  onOpenChange: () => null,
+});
+
+type Props = {
+  children: ReactNode;
+  value?: string;
+  defaultValue?: string;
+  updateValueNode: (value?: string) => ReactElement | null;
+  defaultOpen?: boolean;
+  onOpenChange?: (value: boolean) => void;
```

Comment:
I find this name a bit strange. Maybe `onToggleOpen(value: 'open' | 'closed')`

Reply:
This is the name used by the radix-ui and that is why used it